### PR TITLE
docs: README.md

### PR DESCRIPTION
**Pull request by alvinhui**
- 简化特性描述；
- 调整章节顺序；
- 添加 withStore API。

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,178 +24,133 @@ npm install @ice/store --save
 
 `icestore` is a lightweight React state management library based on hooks. It has the following core features:
 
-* **Minimal API**: Contains 5 APIs, which is easily learnable in 5 minutes.
-* **Typescript Support**: Provide complete type definitions to support IntelliSense in VSCode.
-* **Predictable**: Uses unidirectional data flow (similar to Redux) and allows state mutation only inside actions, allowing data flow to be traced easily.
-* **Optimal performance**: Decreases the number of view components that rerender when the state changes by creating multiple stores.
-* **Built in async status**: Records loading and error status of async actions, simplifying the rendering logic in the view layer.
+* **Minimal & Familiar API**: No additional learning costs, just use React Hooks as intended.
```
Comment (temper357) [inline README.md]: No additional learning costs, easy to get started with the knowledge of React Hooks.

File: README.md [hunk 2]
```diff
@@ -24,178 +24,133 @@ npm install @ice/store --save
...
+* **Enough & Extensible**: Meet 80% of scenarios, another 20%? Middleware mechanism can be used to meet the requirements.
```
Comment (temper357) [inline README.md]: Cover 80% most common usage scenes and extend the remaining  20% scenes with builtin middleware mechanism.

File: README.md [hunk 3]
```diff
@@ -24,178 +24,133 @@ npm install @ice/store --save
...
+* **Class Component Support**: Friendly compatibility strategy can make the old project enjoy the fun of lightweight state management.
```
Comment (temper357) [inline README.md]: Make old projects enjoying the fun of lightweight state management with friendly compatibility strategy.

**Comment by temper357** [comment]
Reviewed.

File: README.md (hunk 1 above)
Comment (alvinhui) [reply README.md]: done

File: README.md (hunk 2 above)
Comment (alvinhui) [reply README.md]: done

File: README.md (hunk 3 above)
Comment (alvinhui) [reply README.md]: done

**Comment by alvinhui** [comment]
done

**Pull request merged**